src/main/ruby/core/io.rb

```diff
@@ -352,6 +356,17 @@ def inspect # :nodoc:
       format '#<IO::InternalBuffer:0x%x total=%p start=%p used=%p data=%p %s>',
              object_id, @total, @start, @used, @storage, content
     end
+
+    Truffle::Boot.delay do
+      if Truffle::Boot.get_option('polyglot.stdio')
+        def fill_read(io, count)
+          buffer = Truffle::POSIX.read_string_blocking(io, count)
+          bytes_read = buffer ? buffer.bytesize : 0
+
+          [buffer, bytes_read]
+        end
+      end
+    end
```

Comment:
Could you move it right below `fill_read` above so the possible redefinition stays together?

Reply:
Sure. I just put it at the bottom of the class so all such redefinitions would be grouped together.

Reply:
That works too :)